Sources/Subprocess/Subprocess.swift

```diff
@@ -26,147 +26,628 @@
+    ///   - standardInput: A type conforming to `DataProtocol` (typically a `Data` type) from which to read input to the external command.
+    ///   - options: Options used to specify runtime behavior.
+    ///   - decoder: A `TopLevelDecoder` that will be used to decode the data.
+    @inlinable
+    public static func value<Content, Decoder>(for command: [String], standardInput: (any DataProtocol)? = nil, options: RunOptions = .throwErrorOnNonZeroExit, decoder: Decoder) async throws -> Content where Content : Decodable, Decoder : TopLevelDecoder, Decoder.Input == Data {
+        try await decoder.decode(Content.self, from: data(for: command, standardInput: standardInput, options: options))
+    }
+    
+    /// Retreive output from from running an external command.
+    /// - Parameters:
+    ///   - command: An external command to run with optional arguments.
+    ///   - standardInput: A `String` from which to send input to the external command.
+    ///   - options: Options used to specify runtime behavior.
+    ///   - decoder: A `TopLevelDecoder` that will be used to decode the data.
+    @inlinable
+    public static func value<Content, Decoder>(for command: [String], standardInput: String, options: RunOptions = .throwErrorOnNonZeroExit, decoder: Decoder) async throws -> Content where Content : Decodable, Decoder : TopLevelDecoder, Decoder.Input == Data {
+        try await decoder.decode(Content.self, from: data(for: command, standardInput: standardInput, options: options))
+    }
+    
+    /// Retreive output from from running an external command.
+    /// - Parameters:
+    ///   - command: An external command to run with optional arguments.
+    ///   - standardInput: A file `URL` from which to read input to the external command.
+    ///   - options: Options used to specify runtime behavior.
+    ///   - decoder: A `TopLevelDecoder` that will be used to decode the data.
+    @inlinable
+    public static func value<Content, Decoder>(for command: [String], standardInput: URL, options: RunOptions = .throwErrorOnNonZeroExit, decoder: Decoder) async throws -> Content where Content : Decodable, Decoder : TopLevelDecoder, Decoder.Input == Data {
+        try await decoder.decode(Content.self, from: data(for: command, standardInput: standardInput, options: options))
+    }
+}
+
+extension Subprocess {
+    /// Errors specific to `Subprocess`.
+    public enum Error: LocalizedError {
+        case nonZeroExit(status: Int32, reason: Process.TerminationReason, standardOutput: Data, standardError: String)
+        
+        public var errorDescription: String? {
+            switch self {
+            case let .nonZeroExit(status: terminationStatus, reason: _, standardOutput: _, standardError: errorString):
+                return "Process exited with status \(terminationStatus): \(errorString)"
+            }
+        }
+    }
+}
+
+private actor TerminationContinuation {
```

Comment:
Should this `TerminationContinuation` type move to its own file?

Reply:
It's private to that file and specific to that file scope so I don't think it makes sense to move it out since it would lose its private scope.